nav2_route/include/nav2_route/edge_scorer.hpp

```diff
@@ -63,7 +66,8 @@ class EdgeScorer
    * @return If edge is valid
    */
   bool score(
-    const EdgePtr edge, const geometry_msgs::msg::PoseStamped & goal_pose, bool final_edge,
+    const EdgePtr edge, const geometry_msgs::msg::PoseStamped & goal_pose, bool start_edge,
```

Comment:
If we're going to add things like this, I think we should create an `enum` so more than be added without breaking API in the future. We can have an enum for `SpecialEdgeType` with start, final as current ones. If neither, then the default can be `NONE`.